src/sentry/sentry_metrics/consumers/last_seen_updater.py

```diff
@@ -52,6 +53,47 @@ def should_drop(self, message: Message[KafkaPayload]) -> bool:
         return FetchType.DB_READ.value not in str(header_value)
 
 
+class KeepAliveMessageFilter(StreamMessageFilter[TPayload]):
+    """
+    A message filter that wraps another message filter, and ensures that at
+    most `consecutive_drop_limit` messages are dropped in a row. If the wrapped
+    `inner_filter` drops `consecutive_drop_limit` messages in a row, the next
+    message will be unconditionally accepted.
+
+    The reason to use this has to do with the way Kafka works. Kafka works with
+    offsets of messages which need to be committed to the broker to acknowledge
+    them. In cases where there is a shared topic, and only one type of messages
+    on the topic, if we drop all messages then that consumer would never commit
+    offsets to the broker which would result in increase in consumer group lag
+    of the consumer group.
+
+    This leads to false positives in our alerts regarding consumer group having
+    some issues.
+
+    Note: This filter can only be used if the wrapped filter is not required
+    for correctness.
+    """
+
+    def __init__(
+        self, inner_filter: StreamMessageFilter[TPayload], consecutive_drop_limit: int
```

Comment:
Not a big concern for now but I believe `inner_filter` should be optional. It is not necessary that all filtering happens with headers. It is quite possible for the message processors to filter the messages if the needed information for filtering is not available on the headers.

Reply:
but the problem that this wrapper is supposed to solve is caused by having a filter at all. if there was no messagefilter configured, this wrapper isn't useful...